### PR TITLE
Adding IBM Power and Z as valid platforms to the docker

### PR DESCRIPTION
**Pull request by brahmaroutu**
@blackjlack Please see if this can be merged for IBM to allow linux distros to build their Docker binaries.

### CODE REVIEW

**Comment by jfrazelle** [comment]
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

`git commit --amend -s --no-edit && git push -f`

**Comment by tiborvass** [comment]
Ping @unclejack

**Comment by jfrazelle** [comment]
oooooo la la

**Comment by duglin** [comment]
I believe it should be:  ooooo la la   https://www.youtube.com/watch?v=mnEQ6CNvPSg

**Comment by brahmaroutu** [comment]
sorry, all happened so fast before I realized you guys are awesome. Please expedite this PR if possible.

**Comment by jfrazelle** [comment]
:) lol

File: daemon/daemon.go [hunk 1]
```diff
@@ -1207,7 +1207,7 @@ func (daemon *Daemon) ImageGetCached(imgID string, config *runconfig.Config) (*i
 
 func checkKernelAndArch() error {
 	// Check for unsupported architectures
-	if runtime.GOARCH != "amd64" {
+	if runtime.GOARCH != "amd64" && runtime.GOARCH != "ppc64" && runtime.GOARCH != "ppc64le" && runtime.GOARCH != "s390x" {
 		return fmt.Errorf("The Docker runtime currently only supports amd64 (not %s). This will change in the future. Aborting.", runtime.GOARCH)
```
Comment (estesp) [inline daemon/daemon.go]: Shouldn't the message be changed as well?  But, not sure it needs to list everything that is (going to be) supported; maybe "The Docker runtime is not enabled for architecture %q."

**Comment by crosbymichael** [comment]
Going forward I would be inclined to just remove this check all together.  It's one of the first things that people remove if they are working on different arches and i'm not sure i see much value in this check.  

If your arch is not supported then there probably are not any binaries available for it.  What do the rest of the maintainers think?

**Comment by duglin** [comment]
@crosbymichael sounds reasonable to me

**Comment by brahmaroutu** [comment]
Taking out the check altogether would make sense. PR updated.

**Comment by estesp** [comment]
Sorry to add a nit, but seems it would make sense to no longer call the function `checkKernelAndArch`.  How about changing to just `checkKernel`?

**Comment by unclejack** [comment]
@brahmaroutu Can you also make the change as @estep suggested, please? I'm OK with this change, but we've got to make the other change as well so we don't have to do it in another commit and PR.

**Comment by tianon** [comment]
+1 to removing the arch check

**Comment by paultag** [comment]
We patch that check out in Debian, :+1: to removing it wholesale, it's pretty easy to remove anyway

**Comment by brahmaroutu** [comment]
Renamed the call as per the suggestion.

**Comment by tianon** [comment]
LGTM

**Comment by duglin** [comment]
LGTM

**Comment by unclejack** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Comment by icecrime** [comment]
I'm fine adding this to 1.5.0 is there's a strong interest for it.

**Comment by brahmaroutu** [comment]
It would be awesome if it can go into 1.5. Canonical is building Docker binaries on Ubuntu15.04 for IBM Power systems. We will have one less workaround to do that task. Kindly see if this can go into 1.5

**Comment by icecrime** [comment]
@brahmaroutu It will :+1: Thanks!

**Pull request merged**